### PR TITLE
Add shi7

### PR DESCRIPTION
**Pull request by lebovic**
Adds:
- `shi7` tool
- `shi7` single-end integration test
- `shi7` paired-end integration test, but it's non-deterministic so it just confirms it's not the same as single-end

Inputs are artificially constrained at 10 for now.

### CODE REVIEW

File: toolchest_client/tools/api.py [hunk 1]
```diff
@@ -109,8 +109,8 @@ def kraken2(output_path, inputs=[], database_name="standard", database_version="
 
     if read_one:
         inputs = [read_one]
-    if read_two:
-        inputs.append(read_two)
+        if read_two:
+            inputs.append(read_two)
```
Comment (lebovic) [inline toolchest_client/tools/api.py:113]: Not related to shi7, but formerly the R2 (only) could be added in paired-end mode.

File: toolchest_client/tools/tool.py [hunk 2]
```diff
@@ -73,12 +73,11 @@ def _validate_args(self):
...
         if self.output_path is None:
             raise ValueError("No output path provided.")
-        try:
-            with open(self.output_path, "a") as _:
-                pass
-        except OSError:
+        if not os.access(
+                os.path.dirname(self.output_path),
+                os.W_OK | os.X_OK,
+        ):
```
Comment (lebovic) [inline toolchest_client/tools/tool.py:79]: Prevents creating an empty file, such that:
```
toolchest.shi7(
    inputs="./directory/",
    output_path="./directory/,
)
```

works.

File: toolchest_client/tools/api.py (hunk 1 above)
Comment (bcai2) [reply toolchest_client/tools/api.py:113]: We might want to add a note to the docstring about this, but I think that the docs/docstrings are getting unwieldly and may in need of restructuring soon, so I'm fine with specifying this later.

File: toolchest_client/tools/api.py [hunk 3]
```diff
@@ -135,6 +135,32 @@ def kraken2(output_path, inputs=[], database_name="standard", database_version="
     instance.run()
 
 
+def shi7(inputs, output_path, tool_args=""):
+    """Runs shi7 via Toolchest.."
```
Comment (bcai2) [inline toolchest_client/tools/api.py:139]: typo at the end of this line?

**Review by bcai2 — APPROVED** [review]
LGTM; found a typo(?) and made a note about docs, but ready to merge when you are :+1:

File: toolchest_client/tools/api.py (hunk 3 above)
Comment (lebovic) [reply toolchest_client/tools/api.py:139]: Whoops, yep.

**Pull request merged**